BFS_Connected_Components/Q10_Leetcode_number_of_closed_islands.py

```diff
@@ -0,0 +1,38 @@
+import collections
+def closedIsland(grid):
+    ROWS, COLS = len(grid), len(grid[0])
+    
+    directions = [[0, 1], [0, -1], [1, 0], [-1, 0]]
+    seen = set()
+    
+
+    def is_closed_island(r, c):
+        q = collections.deque([(r, c)])
+        seen.add((r, c))
+        closed = True
+        while q:
+            row, col = q.popleft()
+            is_boundary = row == ROWS - 1 or row == 0 or col == 0 or col == COLS - 1
+            if is_boundary:
+                    closed = False
+            for dr, dc in directions:
+                r, c = row + dr, col + dc
+                is_in_bounds = r in range(ROWS) and c in range(COLS)
+                is_not_in_seen = (r, c) not in seen
+                if is_in_bounds and is_not_in_seen and grid[r][c] == 0:
```

Comment:
What does 0 mean? In this problem, 0 means land. So, create a constant at the top of the file called `LAND` and use that here.